frontend/deb/handle_debroot.go

```diff
@@ -0,0 +1,148 @@
+package deb
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+)
+
+func DebrootHandler(target string) frontend.BuildFunc {
+	return func(ctx context.Context, client gwclient.Client, spec *dalec.Spec) (gwclient.Reference, *frontend.Image, error) {
+		st, err := Debroot(spec, llb.Scratch(), target, "")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		sOpt, err := frontend.SourceOptFromClient(ctx, client)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		sm, err := dalec.Sources(spec, sOpt)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for k, src := range sm {
+			sm[k] = llb.Scratch().File(llb.Copy(src, "/", k, dalec.WithCreateDestPath()))
+		}
+
+		sources := maps.Values(sm)
+		st = dalec.MergeAtPath(llb.Scratch(), append(sources, st), "/")
+
+		def, err := st.Marshal(ctx)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "error marshalling llb")
+		}
+
+		res, err := client.Solve(ctx, gwclient.SolveRequest{
+			Definition: def.ToPB(),
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ref, err := res.SingleRef()
+		if err != nil {
+			return nil, nil, err
+		}
+		return ref, &frontend.Image{}, nil
+	}
+}
+
+// Debroot creates a debian root directory suitable for use with debbuild.
+// This does not include sources in case you want to mount sources (instead of copying them) later.
+func Debroot(spec *dalec.Spec, in llb.State, target, dir string) (llb.State, error) {
+	control, err := Dalec2ControlLLB(spec, in, target, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating control file")
+	}
+
+	rules, err := Rules(spec, in, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating rules file")
+	}
+
+	changelog, err := Changelog(spec, in, target, dir)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error generating changelog file")
+	}
+
+	if dir == "" {
+		dir = "debian"
+	}
+	compat := llb.Scratch().
+		File(llb.Mkdir(dir, 0o755, llb.WithParents(true))).
+		File(llb.Mkfile(filepath.Join(dir, "compat"), 0o770, []byte("10")))
```

Comment:
I see this is [recommended to be hardcoded](https://www.debian.org/doc/manuals/maint-guide/dother.en.html#compat). How about a `const`?

Reply:
for `10` and `compat` both, I guess